### ISSUE TITLE
ComposeRedundantComposable incorrectly suggests removing @Composable

### ISSUE THREAD

**Issue by nuhkoca** [slackhq/compose-lints#574]
Hello,

`ComposeRedundantComposable` reports correctly in most cases, but I noticed one scenario that is not handled properly. The IDE suggests removing the `@Composable` annotation, even though the function contains a composable call to `BoxWithConstraints`.

It seems this case is being incorrectly identified as redundant.

<img width="961" height="932" alt="Image" src="https://github.com/user-attachments/assets/56482b2a-09f5-4826-aa84-877cdec8811c" />

<img width="504" height="116" alt="Image" src="https://github.com/user-attachments/assets/d1b38db3-8bcd-49ad-bcfa-b4ce2c2cc49b" />

**Comment by ZacSweers** [comment]
Like your other issues, this is not enough information. Please supply more information and a minimally reproducible sample

**Comment by ahmedre** [comment]
i've seen this recently and this (admittedly super contrived example) repros it for me (just simplified an existing function that had this) -

```kotlin
enum class SampleState {
   ONE, TWO
}

@Composable
fun rememberCollapsed(
    state: SampleState,
    contentHeight: Dp
): State<Dp> {
    return remember(contentHeight) {
        derivedStateOf {
            when (state) {
                SampleState.ONE -> 84.dp
                SampleState.TWO -> 88.dp
            }
        }
    }
}
```

**Comment by hiberfake** [comment]
I can confirm that as well. Since version **1.5.x**, it reports `ComposeRedundantComposable`.
In my case, it's the following code snippet:

```kotlin
val MaterialTheme.elevations: Elevations
    @Composable
    @ReadOnlyComposable
    get() = LocalElevations.current
```

<img width="1970" height="720" alt="Image" src="https://github.com/user-attachments/assets/8c62b255-06df-41d3-a91b-65b513b53042" />

**Comment by jgavir23** [comment]
This is a easy way to reproduce this:
``` kotlin
@Composable
internal fun ProgressSlider(
    stepsCount: Int,
    modifier: Modifier = Modifier
) {
    Row(
        modifier = modifier,
        verticalAlignment = Alignment.CenterVertically
    ) {
        for (i in 0 until stepsCount) {
            Spacer(
                modifier = Modifier.padding(end = if (i < stepsCount - 1) 24.dp else 0.dp)
            )
        }
    }
}
```

I'm using version `1.5.2` of the library and Android Studio Quail 1 | 2026.1.1 Patch 2

<img width="1014" height="373" alt="Image" src="https://github.com/user-attachments/assets/bef0fed0-0671-434d-9faf-f62efecfc1b1" />

**Comment by ZacSweers** [comment]
Hi folks, I have not been able to reproduce this in a unit test or project. Can someone please link a minimally reproducing sample?

**Comment by jgavir23** [comment]
@ZacSweers Sure, here is a minimal example for that on [here](https://github.com/jgavir23/slack-compose-lint-test). You can go to `app/src/main/java/com/example/myapplication/ProgressSlider.kt` and you will get this:

<img width="1042" height="304" alt="Image" src="https://github.com/user-attachments/assets/0420b1e8-3ee2-48dd-8335-0b8200b27142" />

**Comment by veyndan** [comment]
I think one thing that hasn't been mentioned explicitly here is that this error seems to _only_ occur in Android Studio. Running `./gradlew lint` doesn't give these false positives and seems to be working correctly. The issue is only in the IDE.
